src/main/fc/config.c

```diff
@@ -410,6 +411,13 @@ static void validateAndFixConfig(void)
 #ifndef USE_OSD_SLAVE
 void validateAndFixGyroConfig(void)
 {
+#ifdef USE_GYRO_DATA_ANALYSE
+    // Disable dynamic filter if gyro loop is less than 2KHz
```

Comment:
I think we should check for `gyro.targetLooptime <= HZ_TO_INTERVAL_US(2000))` here, and then only check for `feature(FEATURE_DYNAMIC_FILTER)` in `gyro.c`. This is how it is done for everything else.

Reply:
Note there is a chance the the RC could wedge on the first boot after changing the gyro loop to be less than 2K.  This is because the `targetLooptime` is not available until after the gyro loop starts and disabling the feature at that point will only make it inactive for the next reboot (because of the "latching" effect of features).  This is what I was asking about in slack.

Reply:
Ah I see. I think we calculate the target gyro loop time as part of calculating the target PID loop time earlier on in `validateAndFixConfig()`, so we could potentially piggy back on this.

Reply:
The problem is primarily that the feature change doesn't take effect until the next boot because of the "latching".  So I'm not sure doing it earlier would help.

Actually thinking about it overnight I think the risk is very low and not worth worrying about.  The potential problem can only happen on the single boot following the change from > 2K to < 2K.  Since < 2K is such a limited use-case as well it's probably better to simplify back to the feature check-only rather than incurring the performance hit in the 99.99% case.

Reply:
I think the latching is reset after the `validateConfig()` on EEPROM load. Also, there is #6254 that gets rid of the silly latching altogether.